src/Compilers/VisualBasic/Portable/Semantics/OverloadResolution.vb

```diff
@@ -2866,6 +2866,7 @@ Bailout:
             Debug.Assert(argumentNames.IsDefault OrElse (argumentNames.Length > 0 AndAlso argumentNames.Length = arguments.Length))
             Debug.Assert(Not candidate.ArgumentMatchingDone)
             Debug.Assert(candidate.State = CandidateAnalysisResultState.Applicable)
+            Debug.Assert(Not candidate.Candidate.UnderlyingSymbol.IsReducedExtensionMethod() OrElse methodOrPropertyGroup.ReceiverOpt IsNot Nothing)
```

Comment:
@AlekseyTs This assertion fails (but doesn't indicate a bug). The failure is due to a test calling into the binder directly, and manually creating malformed `BoundMethodGroup`.

To be specific, this is the problematic line (as far as I can tell):

https://github.com/dotnet/roslyn/blob/cd1d0fec5ecf8e23725e95bcf3426b358baf2a59/src/Compilers/VisualBasic/Test/Semantic/Semantics/OverloadResolution.vb#L46-L50

The `receiver` of `BoundMethodGroup` ctor is passed "Nothing", while we're calling into a reduced extension method.

One way to fix that is to change the assert and add `OrElse callerInfoOpt Is Nothing`. But I'm not sure if you'll like that.
Creating dummy receivers is going to be very hard/tedious.
Also, changing the test to "CreateCompilation" and pass a source string is tedious, the test is very very very large.

Any ideas/suggestions?

Reply:
> Any ideas/suggestions?

Perhaps we can check if syntax is from ```VisualBasicSyntaxTree.Dummy```